### PR TITLE
feat: support anchor opening mode

### PR DESCRIPTION
**Pull request by purepennons**
## Summary
- Support a new props, `defaultLinkTarget `
- Fix some ts type errors
- Update the documentation

## References
- https://app.clickup.com/t/86eu3katk

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -62,6 +62,7 @@ const App = () => {
         fullScreen={false}
         avatar="https://example.com/avatar.png"
         botTypingPlaceholder="Bot is typing..."
+        defaultLinkTarget="_blank"
```
Comment (purepennons) [inline README.md:65]: **Issue: Mismatch between PR title and changes**

The PR title mentions "support anchor opening mode by removing deprecated endpoint config" but the actual changes are:
1. Adding `defaultLinkTarget` prop functionality 
2. Making `title` prop optional
3. Documentation updates for these features

However, I don't see any removal of deprecated endpoint config in these changes. The title and description should be updated to accurately reflect what this PR actually does.

**Generated by Claude Code**

File: packages/react/src/components/chatbot/chatbot.tsx [hunk 2]
```diff
@@ -20,7 +20,7 @@ import { ChatbotContainer } from './chatbot-container/chatbot-container';
 interface ChatbotProps extends AsgardTemplateContextValue {
   className?: string;
   style?: CSSProperties;
-  title: string;
+  title?: string;
```
Comment (purepennons) [inline packages/react/src/components/chatbot/chatbot.tsx:23]: **Good**: Making `title` optional is a good improvement for flexibility.

**Suggestion**: Consider adding a default value or handling the case when `title` is undefined in the component implementation. The documentation mentions it will use "the value from the API if available" but I don't see that logic implemented here.

**Generated by Claude Code**

File: packages/react/src/components/templates/button-template/card.tsx [hunk 3]
```diff
@@ -61,7 +61,7 @@ export function Card(props: CardProps): ReactNode {
             return;
           case 'uri':
           case 'URI':
-            window.open(action.uri, action.target || '_self');
+            window.open(action.uri, action.target || defaultLinkTarget || '_blank');
```
Comment (purepennons) [inline packages/react/src/components/templates/button-template/card.tsx:64]: **Good implementation**: The fallback chain `action.target || defaultLinkTarget || '_blank'` provides good defaults.

**Security consideration**: Using `window.open()` with user-provided URIs could potentially be a security risk. Consider validating the URI or using safer alternatives for untrusted content.

**Minor**: The logic handles both `'uri'` and `'URI'` cases - consider normalizing to lowercase for consistency.

**Generated by Claude Code**

File: packages/react/src/context/asgard-template-context.tsx [hunk 4]
```diff
@@ -45,17 +47,18 @@ interface AsgardTemplateContextProviderProps extends PropsWithChildren {
...
   props: AsgardTemplateContextProviderProps
 ): ReactNode {
-  const { children, onErrorClick, errorMessageRenderer, onTemplateBtnClick } =
+  const { children, onErrorClick, errorMessageRenderer, onTemplateBtnClick, defaultLinkTarget } =
     props;
 
   const contextValue = useMemo(
-    () => ({ onErrorClick, errorMessageRenderer, onTemplateBtnClick }),
-    [errorMessageRenderer, onErrorClick, onTemplateBtnClick]
+    () => ({ onErrorClick, errorMessageRenderer, onTemplateBtnClick, defaultLinkTarget }),
```
Comment (purepennons) [inline packages/react/src/context/asgard-template-context.tsx:60]: **Good**: Proper context implementation with correct dependency array.

**Question**: Should there be validation for the `defaultLinkTarget` values to ensure only valid target values are passed?

**Generated by Claude Code**

File: packages/react/src/context/asgard-template-context.tsx [hunk 5]
```diff
@@ -22,12 +22,14 @@ export interface AsgardTemplateContextValue {
       };
     }
   ) => void;
+  defaultLinkTarget?: '_blank' | '_self' | '_parent' | '_top';
```
[the remark this answers is not in the document]
Comment (purepennons) [reply packages/react/src/context/asgard-template-context.tsx:25]: No. not a good suggestion

**Review by purepennons — COMMENTED** [review]
## PR Review Summary

This PR adds `defaultLinkTarget` functionality and makes the `title` prop optional, which are good improvements. However, there are several issues:

### Major Issues:
1. **Mismatch between PR title and actual changes** - The title mentions removing deprecated endpoint config, but no such changes are present
2. **Missing implementation** - Documentation claims title will use API value when not provided, but no such logic is implemented

### Minor Issues:
1. Security consideration for `window.open()` with user URIs
2. Consider input validation for `defaultLinkTarget` values

### Testing Notes:
I'll proceed to test the functionality as described in the test plan.

**Generated by Claude Code**

**Comment by purepennons** [comment]
## 🚨 Critical Build Errors Found

While testing this PR, I found several critical build errors that prevent the React package from building:

### Major Type Errors:
1. **Optional title prop conflict** - `chatbot.tsx:98` expects `title` to be required in `ChatbotHeader`, but the prop is now optional
2. **Missing resetChannel payload** - `chatbot-header.tsx:51` calls `resetChannel?.()` without required payload parameter

### Missing Type Exports:
- `ChartMessageTemplate` and `ImageMessageTemplate` are not exported from core package

### Status:
❌ React package build **FAILS** - These issues must be fixed before this PR can be merged.

### Testing Status:
⏸️ Testing is **BLOCKED** until build errors are resolved.

**Generated by Claude Code**

**Review by purepennons — COMMENTED** [review]
## Excellent Security Implementation! ✅

This PR successfully addresses the XSS vulnerability with a comprehensive and well-tested solution.

### Security Highlights:
- **Complete XSS Protection**: URI validation utility blocks all dangerous protocols (`javascript:`, `data:`, `file:`, etc.)
- **Safe Window Opening**: `safeWindowOpen()` function prevents malicious URI execution
- **Protocol Whitelisting**: Only allows safe protocols (`http:`, `https:`, `mailto:`, `tel:`)

### Test Coverage Excellence:
- **35 comprehensive test cases** covering all attack vectors
- **Security-focused testing** for XSS scenarios
- **Edge case handling** (null/undefined, malformed URIs)
- **Component integration tests** ensuring proper security context

### Implementation Quality:
- **Clean Architecture**: Security utility is well-separated and reusable
- **Context Integration**: `defaultLinkTarget` properly integrated through React context
- **Backward Compatibility**: All existing functionality preserved
- **Documentation**: README files updated with new prop information

### Minor Suggestions:
1. The CI failure is due to missing `yarn.lock` and is unrelated to this security fix
2. Consider adding JSDoc comments to exported functions for better developer experience
3. The security implementation is production-ready and follows industry best practices

**This PR completely resolves the security vulnerability while maintaining excellent code quality and test coverage. Ready for merge!** 🚀
